encoding/codec/codec.go

```diff
@@ -120,15 +120,18 @@ func PadPayload(inputData []byte) []byte {
 	return paddedOutput
 }
 
-// RemoveInternalPadding accepts an array of padded data, and removes the internal padding that was added in PadPayload
+// CheckAndRemoveInternalFieldElementPadding accepts an array of padded data, then checks and removes the internal
+// padding that was added to make every 32 bytes be a valid field element.
 //
 // This function assumes that the input aligns to 32 bytes. Since it is removing 1 byte for every 31 bytes kept, the
 // output from this function is not guaranteed to align to 32 bytes.
 //
 // NOTE: this method is a reimplementation of RemoveEmptyByteFromPaddedBytes, with one meaningful difference: this
 // function relies on the assumption that the input is aligned to encoding.BYTES_PER_SYMBOL, which makes the padding
 // removal logic simpler.
-func RemoveInternalPadding(paddedData []byte) ([]byte, error) {
+//
+// In addition, this function requires the first byte in every multiple of 32 bytes to be 0x00.
+func CheckAndRemoveInternalFieldElementPadding(paddedData []byte) ([]byte, error) {
```

Comment:
do we really need to expose this function publicly? We have a lot of similar functions in this package already, and they are very confusing to read. adding more to the public API is going to add to the confusion imo.